content/blog/r-frictionless-validation.md

```diff
@@ -0,0 +1,89 @@
+
+It also does not support some of our other use cases. The `frictionless` R package can only handle tabular data, not geojson. And the current implementation at the time of writing does not support `missingValues` from the [table schema spec](https://specs.frictionlessdata.io/table-schema/#missing-values) which would be essential in R for working with `NA` values.
+
+## datapackage.r
+
+The second package `datapackage.r` is also developed by the Frictionless Data organisation. This is an older package and looks to be out of active development in favour of new `frictionless` package. It does however offer validation by using the `tableschema.r` package.
+
+This works in a similar way to the `frictionless` package to create and modify Frictionless data packages. Like with `frictionless`, validation against the schema is done when we read the data. If we try to read a data package containing a resource which is missing column we get an error when calling `read()` like
+
+```a
+Error: Table headers don't match schema field names
+```
+
+This also does not look very actionable — which table headers don't match? Are some missing? Are they misspelled? This combined with the fact that the package is not under active development doesn't make this a compelling option.
+
+## Reticulate
+
+Another possibility would be using the Python validation code directly via [reticulate](https://rstudio.github.io/reticulate/). The Python implementation does support a wide range of build in checks and has ability to specify [custom checks](https://framework.frictionlessdata.io/docs/guides/validating-data.html#custom-checks). Additionally, it supports validation for JSON files, and other file types can be supported using plugins.
+
+While Frictionless data packages in Python offer lots of flexibility, reproducing our current suite of validation checks would require a significant amount of effort. 
+
+## Conclusion
+
+After investigating the available R packages for validating Frictionless Data packages, we found they lack the functionality we require. The `frictionless` package, which mainly focuses on package reading and writing, lacks a `validate` function, and its schema checking feature does not produce actionable errors. The `datapackage.r` package is no longer being actively developed and the validation it does have does not always return actionable errors. Using the Frictionless Python package directly would require a significant amount of work to create plugins for handling all file types, develop custom checks that fulfill all validation requirements and build a reticulate wrapper to enable its use from R. If our aim is to have a common code base for validation we could also explore shipping a docker container with a simple API wrapping our R checks which can be deployed anywhere it is needed. This approach would offer the reusability we want while minimising the need for building custom plugins and validation checks in Frictionless.
```

Comment:
```suggestion
After investigating the available R packages for validating Frictionless Data packages, we found they lack the functionality we require. The `frictionless` package, which mainly focuses on package reading and writing, lacks a `validate` function, and its schema checking feature does not produce actionable errors. The `datapackage.r` package is no longer being actively developed and, the validation it does have, does not always return actionable errors. Using the Frictionless Python package directly would require a significant amount of work to create plugins for handling all file types, develop custom checks that fulfill all validation requirements and build a reticulate wrapper to enable its use from R. If our aim is to have a common code base for validation we could also explore shipping a docker container with a simple API wrapping our R checks which can be deployed anywhere it is needed. This approach would offer the reusability we want while minimising the need for building custom plugins and validation checks in Frictionless.
```
I thought that sentence read a little weird. Maybe those commas help break it up?

Reply:
it is not highlighting the changes for me but i have added 2 commas in the sentence that begins with "The `datapackage.r` package"